src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASIC_LENGTH = 10;
+    private T[] list;
+    private int size;
+
+    public ArrayList() {
+        list = (T[]) new Object[BASIC_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (checkSize()) {
+            resize();
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index of list");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        } else if (list.length == size) {
+            resize();
+        }
+        addByIndex(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexIsValid(index);
+        return list[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexIsValid(index);
+        list[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexIsValid(index);
+        T temp = list[index];
+        System.arraycopy(list,index + 1,list,index,size - index - 1);
+        size--;
```

Comment:
Don't forget to remove a stale value:
```suggestion
        list[--size] = null;
```